src/doc/reference.md

```diff
@@ -1178,11 +1178,20 @@ let px: i32 = match p { Point(x, _) => x };
 ```
 
 A _unit-like struct_ is a structure without any fields, defined by leaving off
-the list of fields entirely. Such types will have a single value. For example:
+the list of fields entirely. Such structure implicitly defines a constant of
+its type with the same name. For example:
```

Comment:
1. I like this way of phrasing the semantics better than the RFC text itself (which had that odd "flexible" versus "braced" distinction).  Bravo

2. Nit: I think the correct grammar here would be "Such *a* structure implicitly defines"; but honestly I won't let this *alone* block the PR.  (Let's see where the review goes.)